### PR TITLE
adding options to run notebook with kernel gateway

### LINKED ISSUE #3005: Add option to run Notebook with Kernel Gateway
We propose to make 2 handlers in Jupyter Notebook configurable. Currently, Jupyter Notebook supports a configurable handler for Login. We want to make the following settings configurable at run-time:
	1. 'services.kernels.handlers'
	2. 'services.kernelspecs.handlers'
Additionally, if the Jupyter Community wishes, we could also add the files from the nb2kg demo to the Jupyter Notebook project so that others can use the integration just by changing the configuration. 

These changes will allow users to use Kernel Gateway with Notebook as shown in the nb2kg demo.

### PR DESCRIPTION
**Pull request by shiti-saxena**
resolves #3005

### CODE REVIEW

File: notebook/notebookapp.py [hunk 1]
```diff
@@ -977,6 +982,18 @@ def _update_mathjax_config(self, change):
 
     info_file = Unicode()
 
+    kernel_handlers = Unicode(
+        default_value='services.kernels.handlers',
+        config=True,
+        help=_('The kernel handlers to use.'),
+    )
+
+    kernelspecs_handlers = Unicode(
```
Comment (minrk) [inline notebook/notebookapp.py]: nit: singluar 'kernelspec_handlers' to match 'kernel_handlers'

**Comment by shiti-saxena** [comment]
@minrk It is possible to modify the `load_handlers` method to allow overriding any handler as currently it adds `notebook.` prefix. But I wasn't sure if we would want to expose that to users. Should I update the PR to do that as well?

**Comment by shiti-saxena** [comment]
@minrk @parente I've created another PR https://github.com/jupyter/notebook/pull/3158 based on the feedback and comments, what do you think of it?

**Comment by takluyver** [comment]
Closing for now in favour of #3158

**Pull request closed without merging**